static/js/components/LearningResourceCard.js

```diff
@@ -191,24 +187,8 @@ export const LearningResourceCard = ({
 }
 
 const mapDispatchToProps = dispatch => ({
-  toggleFavorite: payload => {
-    if (payload.object_type === LR_TYPE_COURSE) {
-      dispatch(mutateAsync(favoriteCourseMutation(payload)))
-    }
-    if (payload.object_type === LR_TYPE_BOOTCAMP) {
-      dispatch(mutateAsync(favoriteBootcampMutation(payload)))
-    }
-    if (payload.object_type === LR_TYPE_PROGRAM) {
-      dispatch(mutateAsync(favoriteProgramMutation(payload)))
-    }
-    if (
-      [LR_TYPE_USERLIST, LR_TYPE_LEARNINGPATH].includes(payload.object_type)
-    ) {
-      dispatch(mutateAsync(favoriteUserListMutation(payload)))
-    }
-    if (payload.object_type === LR_TYPE_VIDEO) {
-      dispatch(mutateAsync(favoriteVideoMutation(payload)))
-    }
+  showListDialog: payload => {
+    dispatch(setDialogData({ dialogKey: DIALOG_ADD_TO_LIST, data: payload }))
```

Comment:
this would be a bit of a refactor from your current setup, but what if instead of dispatching this `setDialogData` action to control the the list dialog we just had a `useState` call here on the `LearningResourceCard`?

You could basically have this:

```js
const [showAddToListDialog, setShowAddToListDialog] = useState(false)

return (
    ...
    { showAddToListDialog ? (
      <AddToListDialog
        object={object}
        hide={() => setShowAddToListDialog(false)}
      />
    ) : null }
)
```

I think that would simplify a few things, namely you wouldn't need to have the `objectType` and `objectId` values stored in the store, and the corresponding logic in the `AddToListDialog` selectors for pulling the object out based on those values. You also wouldn't need to dispatch the dialog actions to open and close the dialog.

Reply:
actually if you did that this `LearningResourceCard` component also wouldn't even need to have the `connect` call any longer

Reply:
@aliceriot if I put `<AddToListDialog>` inside `LearningResourceCard`, on the learning home page, the dialog is squished to fit into the dimensions of the carousel:

![Screen Shot 2019-11-04 at 1 31 11 PM](https://user-images.githubusercontent.com/187676/68147206-97508000-ff07-11e9-99a3-c46dd58cec47.png)

Formerly `AddToListDialog` was added to `CourseIndexPage` and `CourseSearchPage`

Reply:
Also, the dialog isn't hidden unless I click within the `<LearningResourceCard>` area (on the learning home page only; not a problem on the search page).

Reply:
hm, that seems like a styling issue with how the course carousel renders things - I can take a look to see if there's a quick workaround if you push that code up

Reply:
@aliceriot I just pushed the latest code

Reply:
just took a look, so the issue is that the carousel has  the `transform` css property set on several elements, which messes up the `fixed` positioning used by the dialog. see the details here: https://developer.mozilla.org/en-US/docs/Web/CSS/position#Syntax

pretty annoying, and there doesn't seem to be a good fix for this unfortunately, since the carousel relies on the transform for changing which items are shown when the user pages through

Reply:
yeah, I tried a few little workarounds and can't find anything that works well, so we might have to use the original strategy instead, unfortunately. probably we don't want to try to change out the course carousel for a different carousel component right now, haha